src/c++/perf_analyzer/test_request_rate_manager.cc

```diff
@@ -619,4 +658,109 @@ TEST_CASE(
   CHECK(recorded_inputs[3][0].second == 4);
 }
 
+/// Check that the using_shared_memory_ is being set correctly
+///
+TEST_CASE("Check setting of InitSharedMemory")
+{
+  PerfAnalyzerParameters params;
+  bool is_sequence = false;
+  bool is_decoupled = false;
+  bool use_mock_infer = true;
+
+  SUBCASE("No shared memory")
+  {
+    params.shared_memory_type = NO_SHARED_MEMORY;
+    TestRequestRateManager trrm(
+        params, is_sequence, is_decoupled, use_mock_infer);
+    CHECK(false == trrm.using_shared_memory_);
```

Comment:
Just curious, why put the literal value first in the comparison? It kinda disrupts the flow that normal English conversation would order things.

Reply:
For 2 reasons:

1. I always reads these types of checks as expected == actual, so i was following that format.
2. You cannot assign to false. If I have a typo, trrm.usuing_shared_memory_ = false will assign and not throw a compilation error. Constructing it the other way, the compiler will catch that typo for me.